### DISCUSSION TITLE
Error on CMake

### DISCUSSION THREAD

**Discussion by daniele-baccega** [FLAMEGPU/FLAMEGPU2#1112] [Q&A] [answered]
Hello,
I've just obtained this error during the compilation and I have no idea on how to solve it:

```
CMake Error at build/_deps/flamegpu2-src/src/CMakeLists.txt:417 (source_group):
  source_group ROOT:
  /home/trustalert/Desktop/cottolengo-hospital-flame-gpu-2/build/_deps/flamegpu2-src
  is not a prefix of file:
  /home/trustalert/Desktop/cottolengo-hospital-flame-gpu-2/build/_deps/flamegpu2-build/FLAMEGPU/src/flamegpu/version.cpp
```

Thank you again for your help

**Comment by Robadob** [comment]
It's possible that the merge of #1062 has broken something, as this was messing around with source trees.

You might be able to fix it by regenerating your build directory, otherwise I would suggest pinning `FLAMEGPU_VERSION` within cmake to 71a0501f9e5a957f919f5391df9c1bb318dbb50c, which predates that PR.

@ptheywood knows the CMake stuff far better than I, so he'll be the one to give you a clearer response in the near future (He's just gone on leave until Wednesday).

**Reply by daniele-baccega** [reply]
I tried to regenerate the build directory, but it fails the same. Trying to use `FLAMEGPU_VERSION` equals to `71a0501` fails with a `fatal: invalid reference: 71a0501`

**Reply by Robadob** [reply]
@ptheywood is looking into it now, will hopefully have a fix (or revert) soon.

**Comment by ptheywood** [comment, marked as answer]
I've addressed this in https://github.com/FLAMEGPU/FLAMEGPU2/pull/1113, which I have now merged into `master`. I've only tested this behaves under windows, but I expect linux should be fine. 

You'll need to reset your `FLAMEGPU_VERSION` to `master`, and a fresh build directory might be needed (but shouldn't). 

Unfortunately during #1062 I overlooked the fact that the dynamic version.cpp file within the FLAMEGPU target is not a child of the flamegpu2 repo clone when fetched via CMake.

 > I tried to regenerate the build directory, but it fails the same. Trying to use FLAMEGPU_VERSION equals to 71a0501 fails with a fatal: invalid reference: 71a0501
 
When pinning against a commit hash, as @Robadob has suggested, you also need to set `FLAMEGPU_VERSION_ALLOW_HASH` to `ON` in your models CMakeLists.txt. This is not possible by default to make fetching FLAME GPU 2 much more lightweight (faster, uses less network and disk storage).

I.e uncomment the following line from the [FLAMEGPU2-model-template-cpp](https://github.com/FLAMEGPU/FLAMEGPU2-model-template-cpp/blob/c7c915e483ecbffa140881aad1228f329863a8c1/CMakeLists.txt#L7C1-L8C1)

```cmake
# set(FLAMEGPU_VERSION_ALLOW_HASH "ON")
```

**Comment by daniele-baccega** [comment]
Thank you very much!
